django_loci/base/admin.py

```diff
@@ -289,6 +295,14 @@ def clean(self):
         fields = []
         if not is_mobile and type_ in ['outdoor', 'indoor']:
             fields += ['location_selection', 'name', 'address', 'geometry']
+        # sync location, clean indoor field basis type and delete floorplans if type is not outdoor
+        if location := data.get('location'):
+            location.type = type_
+            if type_ != "indoor":
+                data['indoor'] = None
+                # set floorplans of all objectlocations to None where this location is related
+                location.objectlocation_set.update(floorplan=None)
+                location.floorplan_set.all().delete()
```

Comment:
This will delete all floors associated with an indoor location.

Consider this scenario:
An indoor location has two floors, and two different devices are linked to these locations. When the type of one device is changed, it also affects the type of the other device. I have tested and confirmed this behavior.

@nemesifier, what do you think? Is this expected behavior?

Upon further investigation, I found that the main issue occurs when a floor is linked to an outdoor location (floors should only be linked to indoor locations). This causes an error, which happens specifically when changing the location type through the device. Attaching a video for reference. 

https://github.com/user-attachments/assets/f5b83cde-9944-448f-8efc-632b0bcf68e4

Reply:
I still think, type of location should not be editable, If user want to change type they can select a new location or remove Geographic information.

Reply:
yes @niteshsinha17 i also encountered this issue during initial commits , when a floor references an outdoor location when changed via devicelocation form so I added a restriction basis the one in location form to alert the user of associated floorplans. 
As you said we can approach this two ways:
- deleting all related floorplans (affects other devices associated to the same location), or
- make the type field non editable/ restriction on change for associated floorplans in device location form.

Reply:
Also had a doubt, why arent we deleting objectlocation models on deletion of devices?

Reply:
As we have multiple ways to solve this issue @nemesifier can suggest which one to implement in better ways. Lets wait for his response.